### DISCUSSION TITLE
Index map from mo_coeff matrix to corresponding basis function

### DISCUSSION THREAD

**Discussion by lamthuy** [pyscf/pyscf#2679] [Q&A] [answered]
As far as we know the columns of the mo_coeff correspond to the basis functions use in the HF calculation. Could you please tell whether we can get the map between the column index and the corresponding basis functions used in the given HF calculation?
Thank you!

**Comment by fishjojo** [comment]
`mo_coeff` is defined as $C_{\mu i}$, where $\mu$ is the atomic orbital index and $i$ is the molecular orbital index.

**Comment by lamthuy** [comment]
Thank you for the answer, I may need to rephrase my question. I under  stand that \mu is the atomic orbital. In GTO-3g for example, each atomic orbital is associated to an atom and a specific atomic orbital represented as a weighted linear combination of Gaussian. For example for H2O, we should have 5 atomic orbitals, among those 5 AOs corresponding to 1s, 2s, 2px, 2py, 2pz of the atom O and 2 more AOs coresponding to 1s and 1s of two H atoms. In that case, \mu is the index of the seven AOs, my question is which values of the \mu correspond to  1s, 2s, 2px, 2py, 2pz of the atom O and which values of the \mu correspond to 2 more AOs coresponding to 1s and 1s of two H atoms?

**Comment by fishjojo** [comment]
The AOs are arranged atom by atom and shell by shell.
[`gto.mole.search_ao_nr`](https://pyscf.org/pyscf_api_docs/pyscf.gto.html#pyscf.gto.mole.Mole.search_ao_nr) can help you locate the specific AO.

**Comment by lamthuy** [comment, marked as answer]
Hi I found the following function that list the labels of the AOs, are the labels ordered with the same order of AOs defined in the mo_coeff?

```
# Create your molecular system
mol = gto.M(atom='H 0 0 0; O 0 0 1; H 1 0 0', basis='sto-3g')

# Perform an SCF calculation to get MO coefficients
mf = scf.RHF(mol)
mf.kernel()
me_coeff = mf.mo_coeff
idx = mol.search_ao_nr(1, 1, -1, 2) # O 2px
a = gto.sph_labels(mol)
```
The sph_labels are:
```
['0 H 1s    ', '1 O 1s    ', '1 O 2s    ', '1 O 2px   ', '1 O 2py   ', '1 O 2pz   ', '2 H 1s    ']
```

**Reply by sunqm** [reply]
yes
